### ISSUE TITLE
How to use the sample_weight parameter of SoftmaxCrossEntropyLoss?

### ISSUE THREAD

**Issue by evahlis** [apache/incubator-mxnet#7918]
Hi,
The documentation of the loss function ([here](https://mxnet.incubator.apache.org/api/python/gluon.html?highlight=softmaxcr#mxnet.gluon.loss.SoftmaxCrossEntropyLoss)) says that there is a parameter called sample_weight, presumably allowing reweighing of the loss for each sample in each batch. 

How to use it? I have an imbalanced training data and want to multiply the loss by the inverse of the class frequency.

Thanks!

**Comment by szha** [comment]
You can do:
```python
loss_func = mx.gluon.loss.SoftmaxCELoss()
loss = loss_func(output, label, sample_weight)
```

**Comment by evahlis** [comment]
Yeah that worked for me before. I meant how was this intended to be used? seems that sample_weight is a parameter (symbol) that can be passed to the constructor. Then you need to call .bind on it later?? Some example code for the above would be great.

**Comment by szha** [comment]
`sample_weight` is a parameter that should be passed in when calling the loss. It can be either symbol or ndarray, depending on whether you're training with imperative or symbolic API.

**Comment by evahlis** [comment]
@szha thanks. It seems that it can't be passed by name though. Passing it as in your example above works, but doing loss_func(output, label, sample_weight=sample_weight) doesn't work. Also, it's described in the docs of the constructor. Could use a bit of clarity on the exact usage. Thanks for the quick responses so far.

**Comment by piiswrong** [comment]
loss function is a block and blocks only accept positional arguments to avoid some headaches.

The docs does need to fixed though.
sample_weight shouldn't show up in Parameters. There should be an example explaining how to use it.

**Comment by szha** [comment]
This will be addressed soon in the documentation fix. @evahlis I'd love to close this for now so that we don't exceed the horrendous 2000 open issue mark, if that's OK with you. I will keep you posted on the documentation fix by referencing this thread.

**Comment by evahlis** [comment]
Sure. Thanks!

**Comment by szha** [comment]
@apache/mxnet-committers: This issue has been inactive for the past 90 days. It has no label and needs triage.

For general "how-to" questions, our [user forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) is a good place to get help.

**Comment by gautamkmr** [comment]
Please remove the label: Need Triage
Please add the label: Doc
